sdcm/nemesis.py

```diff
@@ -443,7 +443,15 @@ def _terminate_cluster_node(self, node):
         self.cluster.terminate_node(node)
         self.monitoring_set.reconfigure_scylla_monitoring()
 
-    def disrupt_nodetool_decommission(self, add_node=True):
+    def disrupt_nodetool_decommission(self, add_node=True, is_seed_node=False):
+        if is_seed_node:
+            if len(self.cluster.seed_nodes) < 2:
+                raise UnsupportedNemesis("To running seed decommission the cluster must contains at least 2 seed nodes")
+            if not self.target_node.is_seed:
+                self.target_node = random.choice(self.cluster.seed_nodes)
+            seed_address = ",".join([node.ip_address for node in self.cluster.seed_nodes])
+            [node.patch_scylla_yaml(seed_address=seed_address) for node in self.cluster.nodes]
+
         self._set_current_disruption('Decommission %s' % self.target_node)
         self.cluster.decommission(self.target_node)
         if add_node:
```

Comment:
In this case, we should add the new node and make it as seed node as well.

Reply:
fixed

Reply:
Fixed

Reply:
not exactly, current code still doesn't update other nodes on the new seed node

Reply:
Done